polaris-service/src/test/java/org/apache/polaris/service/admin/PolarisOverlappingTableTest.java

```diff
@@ -0,0 +1,304 @@
+    private String catalogShortName() {
+      StringBuilder result = new StringBuilder();
+      for (char c : catalog().toCharArray()) {
+        if (c == '_') {
+          break;
+        }
+        result.append(c);
+      }
+      return result.toString();
+    }
+
+    @Override
+    public String toString() {
+      return String.format(
+          "extension=%s, catalog=%s, status=%s",
+          extensionName(), catalogShortName(), response.toString());
+    }
+  }
+
+  /* Used to wrap a catalog name, so the TestConfig's final `catalog` field can be updated */
+  protected static class CatalogWrapper {
+    public String catalog;
+
+    public CatalogWrapper(String catalog) {
+      this.catalog = catalog;
+    }
+
+    @Override
+    public String toString() {
+      return catalog;
+    }
+  }
+
+  @BeforeEach
+  public void setup(PolarisConnectionExtension.PolarisToken adminToken) {
+    userToken = adminToken.token();
+    realm = PolarisConnectionExtension.getTestRealm(PolarisServiceImplIntegrationTest.class);
+    defaultCatalog.catalog = String.format("default_catalog_%s", UUID.randomUUID().toString());
+    laxCatalog.catalog = String.format("lax_catalog_%s", UUID.randomUUID().toString());
+    strictCatalog.catalog = String.format("strict_catalog_%s", UUID.randomUUID().toString());
+    List.of(BASE_EXT, LAX_EXT)
+        .forEach(
+            EXT -> {
+              List.of(defaultCatalog, laxCatalog, strictCatalog)
+                  .forEach(
+                      c -> {
+                        StorageConfigInfo config =
+                            FileStorageConfigInfo.builder()
+                                .setStorageType(StorageConfigInfo.StorageTypeEnum.FILE)
+                                .build();
+                        CatalogProperties.Builder propertiesBuilder =
+                            CatalogProperties.builder()
+                                .setDefaultBaseLocation(String.format("%s/%s", baseLocation, c));
+                        if (!c.equals(defaultCatalog)) {
+                          propertiesBuilder
+                              .addProperty(
+                                  PolarisConfiguration.ALLOW_UNSTRUCTURED_TABLE_LOCATION
+                                      .catalogConfig(),
+                                  String.valueOf(c.equals(laxCatalog)))
+                              .addProperty(
+                                  PolarisConfiguration.ALLOW_TABLE_LOCATION_OVERLAP.catalogConfig(),
+                                  String.valueOf(c.equals(laxCatalog)));
+                        }
+                        Catalog catalogObject =
+                            new Catalog(
+                                Catalog.TypeEnum.INTERNAL,
+                                c.catalog,
+                                propertiesBuilder.build(),
+                                System.currentTimeMillis(),
```

Comment:
nit: hardcode timestamp for deterministic testing

Reply:
lots of UUIDs here too ... would be nice to lock down random to a specific seed

Reply:
Agreed on the UUIDs... to be honest, we need a better way of making sure tests & suites don't step on each others' toes. We already use UUIDs in a few places so I feel okay keeping them around for now, but something deterministic / idiomatic would be better. 

If you want, I can replace this with a (seeded) random number, just let me know.

Reply:
I think we should wait till we centralize some of this test code